### DISCUSSION TITLE
Does smart connection read source of the note or rendered version

### DISCUSSION THREAD

**Discussion by iiezhachenko** [brianpetro/obsidian-smart-connections#532] [General]
Hello,
I'm planning to use plugins like Quoth to reuse pieces of notes in other notes instead of copying text.

So this note
<img width="622" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/7869381/98276066-7f50-4eba-9e83-5d4ead83ce35">

Looks like this as a source
````quoth
path: [[Data Backup Best Practices#^9eacae]]
display: inline
```
```quoth
path: [[Data Backup Best Practices#^5f1f68]]
ranges: "**Encrypt " to "ed access."
display: inline
```
```quoth
path: [[Data Backup Best Practices#Data Backup Best Practices]]
ranges: "Perform Fu" to "age space."
display: inline
```
```quoth
path: [[Data Backup Best Practices#^3418e6]]
ranges: "**Automate" to "tant data."
display: inline
```
`

Will Smart Connections pickup the embedded text?

**Comment by brianpetro** [comment]
@iiezhachenko, great question! Avoiding duplicate text is key to a functional system.

Smart Connections does render Dataview queries. So it is possible.

I'm unfamiliar with the Quoth plugin, but I'm sure a similar integration to what's used for Dataview would work.

I've been spending a lot of time on the backend so that "hooking" into various points and processes is easier. This way further integrations, like one with Quoth, will be easier. 

Another thing I need to explore further is seeing if I can hook into Obsidian's main renderer for implementing a more universal solution to note rendering. Preferably a single process that would handle both Dataview and Quoth based on whatever plugins the user has installed.

Thanks for bringing this to my attention!

🌴
